app/pull-gitlab-variables.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+# arguments:
+# - ATLANTIS_GITLAB_TOKEN
+# - HEAD_REPO_OWNER
+# - HEAD_REPO_NAME
+# - ALLOWLIST_FILE e.g. GOOGLE_APPLICATION_CREDENTIALS
+# - ALLOWLIST_ENV_VAR e.g. AWS_ACCESS_KEY_ID,AWS_SECRET_ACCESS_KEY
+# - REPO_REL_DIR e.g environments/dev/cicd/azure
+
+# Variable needed for `glab`
+export GITLAB_TOKEN=${ATLANTIS_GITLAB_TOKEN}
+
+ENV_SCOPE=$(echo $REPO_REL_DIR | sed -nE 's/.*\/(dev|test|prod|staging)\/.*/\1/p')
```

Comment:
I wanted to allow for more flexible directory structure (but at the cost of 'defined' environments), e.g. `environments/azure/dev/...`

Reply:
I was considering adding the same comment, but for example in `gid-infrastrusture` we have additional level which is `aws`/`gcp`/etc - possible that similar use-case would be in other projects

Reply:
agreed that none of the solutions are ideal, but I would think 99% of cases would be `/environment/[env-name]/stack` rather than `/environment/stack/[env-name]`. WDYT?